packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -653,6 +654,11 @@ class DropdownButton<T> extends StatefulWidget {
   /// [ThemeData.textTheme] of the current [Theme].
   final TextStyle style;
 
+  /// The border to use for the drop-down button's [BoxDecoration].
+  ///
+  /// Defaults to a 0.0 width bottom border with color 0xFFBDBDBD.
+  final BorderSide underline;
```

Comment:
Making this property Widget valued instead would add some flexibility without any cost. The widget would replace the Container in:
```
child: Container(
  height: 1.0,
  decoration: const BoxDecoration(
    border: Border(bottom: BorderSide(color: Color(0xFFBDBDBD), width: 0.0))
  ),
),
```

We took a similar tack in https://github.com/flutter/flutter/pull/29572

Reply:
Sounds good, I've changed this field to be Widget valued.